File: android/app/src/main/java/com/demo/ActivityStarterModule.java
package com.demo;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.NonNull;

import com.facebook.react.bridge.Callback;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;

/**
 * Expose Java to JavaScript. Methods annotated with {@link ReactMethod} are exposed.
 */
final class ActivityStarterModule extends ReactContextBaseJavaModule {

    ActivityStarterModule(ReactApplicationContext reactContext) {
        super(reactContext);
    }

    /**
     * @return the name of this module. This will be the name used to {@code require()} this module
     * from JavaScript.
     */
    @Override
    public String getName() {
        return "ActivityStarter";
    }

    @ReactMethod
    void navigateToExample(String data) {
        Activity activity = getCurrentActivity();
        if (activity != null) {
            Bundle b = new Bundle();
            b.putString("DATA", data);
            Intent intent = new Intent(activity, DashboardActivity.class);
            intent.putExtras(b);
            activity.startActivity(intent);
        }
    }

    @ReactMethod
    void getName(@NonNull Callback callback) {
        Activity activity = getCurrentActivity();
        if (activity != null) {
            callback.invoke("Speed : " + AppUtils.getSpeed());
        }
    }
}

File: android/app/src/main/java/com/demo/ActivityStarterModule.kt
package com.demo

import android.app.Activity
import android.os.Bundle
import android.content.Intent
import com.demo.DashboardActivity
import com.demo.AppUtils
import com.facebook.react.ReactPackage
import com.demo.ActivityStarterModule
import com.facebook.react.ReactActivity
import com.demo.R
import android.widget.TextView
import com.facebook.react.modules.core.DeviceEventManagerModule.RCTDeviceEventEmitter
import android.widget.EditText
import com.facebook.react.ReactActivityDelegate
import com.demo.MainApplication
import com.facebook.react.ReactNativeHost
import com.facebook.react.ReactInstanceManager
import com.facebook.react.devsupport.interfaces.DevSupportManager
import com.facebook.react.devsupport.interfaces.DevOptionHandler
import android.widget.Toast
import com.demo.MainActivity
import com.facebook.react.ReactApplication
import com.demo.ActivityStarterReactPackage
import com.facebook.react.bridge.*
import com.facebook.react.shell.MainReactPackage
import com.facebook.soloader.SoLoader

/**
 * Expose Java to JavaScript. Methods annotated with [ReactMethod] are exposed.
 */
internal class ActivityStarterModule(reactContext: ReactApplicationContext?) :
    ReactContextBaseJavaModule(reactContext) {
    /**
     * @return the name of this module. This will be the name used to `require()` this module
     * from JavaScript.
     */
    override fun getName(): String {
        return "ActivityStarter"
    }

    @ReactMethod
    fun navigateToExample(data: String?) {
        val activity = currentActivity
        if (activity != null) {
            val b = Bundle()
            b.putString("DATA", data)
            val intent = Intent(activity, DashboardActivity::class.java)
            intent.putExtras(b)
            activity.startActivity(intent)
        }
    }

    @ReactMethod
    fun getName(callback: Callback) {
        val activity = currentActivity
        if (activity != null) {
            callback.invoke("Speed : " + AppUtils.getSpeed())
        }
    }
}